store/s3store.go

```diff
@@ -0,0 +1,170 @@
+package store
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+const Suffix string = ".json"
+
+type S3Store struct {
+	Bucket string
+	Prefix string
+	svc    s3.S3
+}
+
+// ensure S3Store conforms to Store interface
+var _ Store = &S3Store{}
```

Comment:
any reason not to reference the interface (instead of the particular backend) at the client level so we don't need to do this check?

Reply:
Not really. I copied this code from SSM.

Reply:
ah ok, didn't realize that